### PR TITLE
Update docs to clarify betafix and hotfix branch cut sources

### PR DESCRIPTION
**Pull request by derekblank**
Updates doc references to clarify which tags/branches hotfixes and betafixes should be cut from.

### CODE REVIEW

File: Releasing.md [hunk 1]
```diff
@@ -386,8 +386,8 @@ At the same time there could also be a regular release going on for example for
 
 | Repo             | Cut From                | Branch Name                        | Merging To                                                       |
 | ---------------- | ----------------------- | ---------------------------------- | ---------------------------------------------------------------- |
-| gutenberg        | rnmobile/release_1.11.0 | rnmobile/release_1.11.1            | trunk & (maybe also) rnmobile/release_1.12.0                     |
-| gutenberg-mobile | release/1.11.0          | release/1.11.1                     | trunk & (maybe also) release/1.12.0                              |
+| gutenberg        | rnmobile/release_1.11.0 (tag) | rnmobile/release_1.11.1            | trunk & (maybe also) rnmobile/release_1.12.0                     |
+| gutenberg-mobile | v1.11.1 (tag)          | release/1.11.1                     | trunk & (maybe also) release/1.12.0                              |
```
Comment (dcalhoun) [inline Releasing.md]: If I am not mistaken, this is incorrect. 
```suggestion
| gutenberg-mobile | v1.11.0 (tag)          | release/1.11.1                     | trunk & (maybe also) release/1.12.0                              |
```

File: Releasing.md [hunk 2]
```diff
@@ -416,8 +416,8 @@ At the same time there could also be a regular release, a betafix or even anothe
 
 | Repo             | Cut From                | Branch Name                        | Merging To                                                       |
 | ---------------- | ----------------------- | ---------------------------------- | ---------------------------------------------------------------- |
-| gutenberg        | rnmobile/release_1.11.0 | rnmobile/release_1.11.1            | trunk & (maybe also) rnmobile/release_1.12.1                     |
-| gutenberg-mobile | release/1.11.0          | release/1.11.1                     | trunk & (maybe also) release/1.12.1                              |
+| gutenberg        | rnmobile/release_1.11.0 (tag) | rnmobile/release_1.11.1            | trunk & (maybe also) rnmobile/release_1.12.1                     |
+| gutenberg-mobile | v1.11.1 (tag)          | release/1.11.1                     | trunk & (maybe also) release/1.12.1                              |
```
Comment (dcalhoun) [inline Releasing.md]:
```suggestion
| gutenberg-mobile | v1.11.0 (tag)          | release/1.11.1                     | trunk & (maybe also) release/1.12.1                              |
```

**Review by dcalhoun — COMMENTED** [review]
Noting the tag makes sense to me. 👍🏻

File: Releasing.md (hunk 1 above)
Comment (derekblank) [reply Releasing.md]: Thanks for the catch.

**Review by geriux — APPROVED** [review]
LGTM! Thanks for updating the docs 🚀

**Pull request merged**